detection_rules/schemas/__init__.py

```diff
@@ -231,19 +231,19 @@ def downgrade(api_contents: dict, target_version: str, current_version: Optional
     if current_version is None:
         current_version = current_stack_version()
 
-    current_major, current_minor = Version(current_version)[:2]
-    target_major, target_minor = Version(target_version)[:2]
+    current_major, current_minor = Version.parse(current_version, optional_minor_and_patch=True)[:2]
```

Comment:
```suggestion
    current = Version.parse(current_version, optional_minor_and_patch=True)[:2]
    
    # then latest use current.major and current.minor as needed
```

Reply:
Updated and changed it to rely on semver functionality more.